snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ql/SnomedQueryLabelerRequestBuilder.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.snomed.core.ql;
+
+import com.b2international.snowowl.core.domain.BranchContext;
+import com.b2international.snowowl.core.request.ResourceRequest;
+import com.b2international.snowowl.core.request.ResourceRequestBuilder;
+import com.b2international.snowowl.core.request.RevisionIndexRequestBuilder;
+import com.b2international.snowowl.snomed.core.domain.SnomedConcept;
+
+/**
+ * @since 7.6
+ */
+public final class SnomedQueryLabelerRequestBuilder 
+		extends ResourceRequestBuilder<SnomedQueryLabelerRequestBuilder, BranchContext, String>
+		implements RevisionIndexRequestBuilder<String> {
+
+	private final String expression;
+	private String descriptionType = SnomedConcept.Expand.FULLY_SPECIFIED_NAME; 
+
+	public SnomedQueryLabelerRequestBuilder(String expression) {
+		this.expression = expression;
+	}
+	
+	public SnomedQueryLabelerRequestBuilder setDescriptionType(String descriptionType) {
```

Comment:
The fact that this builder only accepts the literals `SnomedConcept.Expand.FULLY_SPECIFIED_NAME` or `SnomedConcept.Expand.SYNONYM` should be documented somewhere - users of this request might think that it works with a type SCTID.

Reply:
Yes, it will be :) in the javadoc of the `setDescriptionType` method.

Reply:
Done!